scripts/common_embeddings.py

```diff
@@ -42,27 +39,54 @@ def get_file_title(file_path: str) -> str:
         pass
     return title
 
+class FileMetadataProcessor:
+    """Metadata processing callback with memory of unreachable URLS.
+    FileMetadataProcessor keeps a list of processed files,
+    their titles, URLs and if their URLs were reachable.
```

Comment:
What's the point of keeping a record of all the processed files?

Reply:
Possible uses include retrying, final report compilation, easier diagnosis of issues with url derivation.